### ISSUE TITLE
Image size

### ISSUE THREAD

**Issue by Prinstin** [jiny2001/dcscn-super-resolution#36]
Training data is a large part of my own image data set，testing data is another small part of the image data set,is it right?
What is the maximum size of the training image and testing image?
Does the code work for large size(6000-7000) training data?My GPU is GeForce GT 630M （as follows）,I don't know whether I can train the data on my computer.Can you give me some guidance?
If I installed tensorflow-GPU,will it train with GPU by default?and how long it works?

![3](https://user-images.githubusercontent.com/37499633/42032660-dcb5e30e-7b0c-11e8-9912-9173c9254b84.PNG)

**Comment by jiny2001** [comment]
Yep. Using a lot of images for training and using small data for online test sounds really nice.

As long as your memory is enough, I don't think there is a limi. However, I have only trained with up to 2000x2000 image. What you can do is just try it. If there is something wrong, you can divide your training data into smaller images.

However, 1Gbyte memory is quite small. I don't think my 12 layers 196 filters model will work. you must try much smaller model.

I don't know if your GPU is capable to latest tensorflow and CUDA. But usually when you install tensorflow-gpu and other CUDA/driver stuff, it should work with your GPU. Thx.

**Comment by Prinstin** [comment]
Should I try Ver1 model to train?Or use  python train.py --dataset=set5 --layers=5 --filters=32 --use_nin=false --training_images=10000 ?and what is ‘training_images’ for? How to modify the value of ‘training_images’？

**Comment by jiny2001** [comment]
No, ver1 and ver2 has mostly similar memory consumption as long as you chose option args properly. Actually, the parameter you chose at the above comment looks nice starting point to me.
training_images is, the number of patches trained in one epoch. So, usually, it would be better to be same as the number of input training batches. However, it really depends on the image type/variance and model complexity. You can try small value like 1000 and if the result is not enough, you can increase it.

**Comment by Prinstin** [comment]
Is the number of input training batches calculated by the number of training data,each image size and batch size?it is that each training image is to extract patches(with patch size and stride),and  the number of input training batches is sum of them?
If the image size is not an integer multiple of patch size,how does it split?

**Comment by jiny2001** [comment]
1) When build_batch=False(default), we randomly chose patch for every training batch-image. In this case, images under the training data directory are shuffled sequentially. Pick 1 image and then pick specified size-batch image from random position. Then flip the batch horizontally in half probability. Continue this for training_images number.  This option is for large dataset.

2) When build_batch=True, we split images into batch images. Let's say we have a 48x48(batch_image_size) window on the top left of the image. Generate image from this window and then slide the window 24(stride_size) pixels on the right and then generate batch image. Continue to slide the window until the end. If the window exceeds the right image boundary, then move window to 24 pixels down from top-left corner. Repeating it again until the end of the image. 

If you use build_batch=True, the trainer will create a batch images under "batch_data" before training so that you can see how much batch images are in the train data. :)

**Comment by Prinstin** [comment]
Your paper suggests that the datasets for training are yang91 and bsd200,and build an augmented dataset(augment_level = 4), testing dataset is SET5. If I want to train my own model(scale=2,3 or 4) as you provided.
Use 'python augmentation.py --dataset=yang91 --augment_level=4' and 'python augmentation.py --dataset=bsd200 --augment_level=4' to obtain augmented datasets.But how to train with the two augmented datasets?

**Comment by jiny2001** [comment]
you can not specify multiple dataset at a same time.
1) make directory and name it as "yang_bsd"
2) copy image from "yang91" to "yang_bsd"
3) copy image from "bsd200" to "yang_bsd"
4) use augmentation.py to make 4x images of "yang_bsd". Output data directory name would be like "yang_bsd_4".

Please try it.

**Comment by Prinstin** [comment]
using ‘python train.py --dataset=yang_bsd_4’ for 2x
‘python train.py --scale=3 --dataset=yang_bsd_4' for 3x
‘python train.py --scale=4 --dataset=yang_bsd_4' for 4x
is it right?

**Comment by jiny2001** [comment]
yep.
